File: demos/openai/workflow.py
# workflow.py
# OpenAI API Workflow Demo (Python)
# Tim Fraser
#
# This script demonstrates a minimal OpenAI API request using an API key
# loaded from a local .env file.

# 0. Setup #################################

## 0.1 Load packages ############################

import os
import requests
from dotenv import load_dotenv

## 0.2 Load environment variables ############################

load_dotenv()
openai_api_key = os.getenv("OPENAI_API_KEY")

if not openai_api_key:
    raise ValueError("OPENAI_API_KEY not found. Add it to your .env file.")

# 1. Build request #################################

url = "https://api.openai.com/v1/responses"
headers = {
    "Authorization": f"Bearer {openai_api_key}",
    "Content-Type": "application/json",
}
body = {
    "model": "gpt-4o-mini",
    "input": "Hello. Reply exactly with: Model is working.",
}

# 2. Send request #################################

response = requests.post(url, headers=headers, json=body, timeout=60)
response.raise_for_status()
result = response.json()

# 3. Print response #################################

output = result["output"][0]["content"][0]["text"]
print("\nModel response:\n")
print(output)
print("\nDone.\n")

File: demos/openai/workflow.R
# workflow.R
# OpenAI API Workflow Demo (R)
# Tim Fraser
#
# This script demonstrates a minimal OpenAI API request using an API key
# loaded from a local .env file.

# 0. SETUP ###################################

## 0.1 Load packages #################################

library(httr2)
library(jsonlite)
library(purrr)

## 0.2 Load environment variables ####################################

if (file.exists(".env")) {
  readRenviron(".env")
} else {
  warning(".env file not found. Create one with OPENAI_API_KEY=your_key.")
}

openai_api_key = Sys.getenv("OPENAI_API_KEY")
if (openai_api_key == "") {
  stop("OPENAI_API_KEY not found. Add it to your .env file.")
}

# 1. Build request ###################################

url = "https://api.openai.com/v1/responses"
body = list(
  model = "gpt-4o-mini",
  input = "Hello. Reply exactly with: Model is working."
)

# 2. Send request ###################################

res = request(url) %>%
  req_headers(
    "Authorization" = paste0("Bearer ", openai_api_key),
    "Content-Type" = "application/json"
  ) %>%
  req_body_json(body) %>%
  req_method("POST") %>%
  req_perform()

response = resp_body_json(res)

# 3. Print response ###################################

output = response %>% pluck("output", 1, "content", 1, "text")
cat("\nModel response:\n\n")
cat(output)
cat("\n\nDone.\n")
